File: app/src/main/java/io/mgba/UI/Activities/Interfaces/ISettingsPanelView.java
package io.mgba.UI.Activities.Interfaces;

import android.support.v7.preference.PreferenceFragmentCompat;

public interface ISettingsPanelView {
    PreferenceFragmentCompat findFragment(String id);
    void switchFragment(PreferenceFragmentCompat fragment, String tag);
    String getPreference(String key, String defaultValue);
}

File: app/src/main/java/io/mgba/ui/activities/interfaces/ISettingsPanelView.kt
package io.mgba.ui.activities.interfaces


import androidx.preference.PreferenceFragmentCompat

interface ISettingsPanelView {
    fun findFragment(id: String): PreferenceFragmentCompat
    fun switchFragment(fragment: PreferenceFragmentCompat, tag: String)
    fun getPreference(key: String, defaultValue: String): String
}
